File: app/src/main/java/com/miandroidchallenge/ucoppp/miandroidchallenge/di/AppComponent.java
package com.miandroidchallenge.ucoppp.miandroidchallenge.di;

import android.app.Application;
import android.support.v4.app.Fragment;

import com.miandroidchallenge.ucoppp.miandroidchallenge.ui.deliverydetailsfragment.DeliveryDetailsViewModel;
import com.miandroidchallenge.ucoppp.miandroidchallenge.ui.deliverylistfragment.DeliveryFragmentViewModel;
import com.miandroidchallenge.ucoppp.miandroidchallenge.util.module.ApiModuleV2;
import com.miandroidchallenge.ucoppp.miandroidchallenge.util.module.DeliveriesDbModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, ApiModuleV2.class, DeliveriesDbModule.class})
public interface AppComponent {

    void inject(Application application);
    void inject(DeliveryDetailsViewModel deliveryDetailsViewModel);
    void inject(DeliveryFragmentViewModel deliveryFragmentViewModel);
    void inject(Fragment fragment);
}

File: app/src/main/java/com/miandroidchallenge/ucoppp/miandroidchallenge/application/AppComponent.kt
package com.miandroidchallenge.ucoppp.miandroidchallenge.application

import android.app.Application
import android.support.v4.app.Fragment
import com.miandroidchallenge.ucoppp.miandroidchallenge.ui.deliverydetailsfragment.DeliveryDetailsViewModel
import com.miandroidchallenge.ucoppp.miandroidchallenge.ui.deliverylistfragment.DeliveryFragmentViewModel
import com.miandroidchallenge.ucoppp.miandroidchallenge.ui.mainactivity.MainActivity
import com.miandroidchallenge.ucoppp.miandroidchallenge.util.module.ApiModule
import com.miandroidchallenge.ucoppp.miandroidchallenge.util.module.DeliveriesDbModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, ApiModule::class, DeliveriesDbModule::class])
interface AppComponent {
    fun inject(activity: MainActivity)
    fun inject(application: Application)
    fun inject(fragment: Fragment)
    fun inject(deliveryFragmentViewModel: DeliveryFragmentViewModel)
    fun inject(deliveryDetailsViewModel: DeliveryDetailsViewModel)
}